addons/sale/models/sale_order.py

```diff
@@ -63,7 +63,7 @@ def _rec_names_search(self):
         string="Customer",
         required=True, change_default=True, index=True,
         tracking=1,
-        domain="[('company_id', 'in', (False, company_id))]")
+        check_company=True)
```

Comment:
Hi @mathcoutant 

Making partner_id, partner_invoice_id, partner_shipping_id as check_company=True will cause issue as per bellow.

If there is multi company enabled.
Company "YourCompany" and "Company 2"

Both this company have website -> websiteA and websiteB

Now user called Mitchell Admin default company is YourCompany (with allowed company YourCompany and Company 2)
So he can login to Company 2 and place an order.

When user Mitchell Admin try to add product to cart it will give traceback of 

Incompatible companies on records:
- 'S00051' belongs to company 'Company 2' and 'Customer' (partner_id: 'YourCompany, Mitchell Admin') belongs to another company.
- 'S00051' belongs to company 'Company 2' and 'Invoice Address' (partner_invoice_id: 'YourCompany, Mitchell Admin') belongs to another company.
- 'S00051' belongs to company 'Company 2' and 'Delivery Address' (partner_shipping_id: 'YourCompany, Mitchell Admin') belongs to another company.

Can you check this?

Reply:
If a partner is supposed to be shared between companies, its `company_id` field  should be emptied.
By default, the partner of new users is restricted to the company of the user.  Either the user should log out and have a second account for this second company, or its partner should be shared between companies too.

Before the change of this PR, you would have been able to add products to the cart, but there would have been multi-company issues later on in the flow, e.g when paying for your cart:

![image](https://github.com/odoo/odoo/assets/19364378/72bb2c90-dd0f-48ec-a052-473d849eac15)

We're thinking of improving this in the future (maybe show some kind of banner if the customer should log out or contact the enterprise to adapt its configuration).

Reply:
Hi @Feyensv,

Signing up again with other company/website requires to have separate login/email?

End customer might want to use same email for both the website. but odoo do not allow to have two user with same login email id.

Internal user I can understand we can remove company in partner, but how about website users? 
Can you please advice?

Reply:
> Signing up again with other company/website requires to have separate login/email?

As far as I know, yes.  Indeed the customer might want to use the same email for both website (& companies), but it's not supported in a single database
The company can be removed from website users partners too, but new addresses are by default restricted to the website company, so it'd still create issues (and require custom code or manual actions to do the change).

I suppose the best solution would be to share customers created from the ecommerce (and from users), but I doubt we will do that in a stable version :thinking:.
I'll forward the issue to hopefully change that for future versions, but I doubt a miracle solution can be found for stable versions.